### PR TITLE
feat(helm): support existing secrets

### LINKED ISSUE #453: Enable helm chart to use existing secrets
### Component(s)

helm

### Is your feature request related to a problem? Please describe.

The problem with the current helm chart is that you need to provide secret values in plain text in the `values.yaml`. For users who follow a gitops workflow, this is not the ideal solution because the secrets need to be pushed into a repo.

### Describe the solution you'd like

Normally helm chart provide a possibility to define an existing secret name for the chart. In case this value is provided the helm chart does not create its secret resource. 

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by alejandrobelleza**
## Motivation and Context
Support existing secrets to:
*  avoid having sensitive credentials in Cosmo values.yaml.
* Support gitops workflow. ( external-secrets )

This PR closes #453

## TODO

- [ ] Tests or benchmark included
- [ ] Documentation is changed or added on [https://app.gitbook.com/](https://app.gitbook.com/)
- [ ] PR title must follow [conventional-commit-standard](https://github.com/wundergraph/wundergraph/blob/main/CONTRIBUTING.md#conventional-commit-standard)

### CODE REVIEW

File: helm/cosmo/charts/router/CHART.md [hunk 1]
```diff
@@ -22,6 +22,7 @@ WunderGraph Cosmo router.
 | configuration.otelCollectorUrl | string | `""` | The URL of the Cosmo GraphQL OTEL Collector. Should be internal to the cluster. Default to cloud if not set. |
 | deploymentStrategy | object | `{}` |  |
 | existingConfigmap | string | `""` | The name of the configmap to use for the router configuration. The key "config.yaml" is required. |
+| existingSecret | string | `""` | Existing secret containing the graphApiToken |
```
Comment (StarpTech) [inline helm/cosmo/charts/router/CHART.md:25]: Please document that the secret has to be in the same namespace, named like the chart fullname and the keys has to match with current secret.

**Review by StarpTech — CHANGES_REQUESTED** [review]
Thanks for the PR. We should only improve docs around the usage.

**Review by StarpTech — APPROVED** [review]
LGTM

File: helm/cosmo/charts/router/CHART.md (hunk 1 above)
Comment (alejandrobelleza) [reply helm/cosmo/charts/router/CHART.md]: Sure, I've added a _helper function to set the name. And improved the doc.

**Comment by StarpTech** [comment]
@alejandrobelleza you need to run `make docs` to update docs.

**Comment by PeteMac88** [comment]
@alejandrobelleza Thanks for the PR. I was also working on one :D

**Comment by alejandrobelleza** [comment]
> @alejandrobelleza you need to run `make docs` to update docs.

Sure, done

**Comment by PeteMac88** [comment]
Can we merge this PR?

**Comment by PeteMac88** [comment]
@alejandrobelleza  @StarpTech Wouldn't it be better not to create the secrets in the helm chart in case you pass the existingSecret value? I think this will be problematic because when you create the secret through an external secret provider you will get conflicts since the secret in the chart has the same name.

**Comment by StarpTech** [comment]
@PeteMac88 I agree, is that your use case?

@alejandrobelleza Do I read it correctly, that after setting `existingSecret` the old secret is still created? I'd expect that it is not. Similar to how it works in the majority of bnitnami charts https://github.com/bitnami/charts/blob/3a0dfccb3e0eb0ecd344eaae5f53b547135da938/bitnami/nats/templates/secrets.yaml#L6

**Comment by alejandro-belleza-globant** [comment]
> @PeteMac88 I agree, is that your use case?
> 
> @alejandrobelleza Do I read it correctly, that after setting `existingSecret` the old secret is still created? I'd expect that it is not. Similar to how it works in the majority of bitnami charts https://github.com/bitnami/charts/blob/3a0dfccb3e0eb0ecd344eaae5f53b547135da938/bitnami/nats/templates/secrets.yaml#L6

Yes, there was an if there to support secret creation. Sorry during rebase that change got lost. Going to submit a quick fix.

**Comment by alejandrobelleza** [comment]
https://github.com/wundergraph/cosmo/pull/518 - @StarpTech , the fix

**Comment by StarpTech** [comment]
@PeteMac88 does it address your concern too?

**Pull request merged**